src/lib/ClusterScanner/ClusterScanner.ts

```diff
@@ -70,9 +71,16 @@ export class ClusterScanner extends BaseScanner {
         result
           .filter((item: any) => this.eventsList.includes(item.event))
           .filter((item: any) => JSON.stringify(item.returnValues.operatorIds.map((value: any) => +value)) === JSON.stringify(operatorIds))
+          .sort((a: any, b: any) => a.blockNumber - b.blockNumber)  // Sort by blockNumber in ascending order
           .forEach((item: any) => {
-            if (item.blockNumber > biggestBlockNumber) {
+            if (item.blockNumber >= biggestBlockNumber) {
+              const previousBlockNumber = biggestBlockNumber;
```

Comment:
It looks like you can use the same `latestBlockNumber` variable for both purposes